helix-term/src/commands.rs

```diff
@@ -3650,8 +3650,7 @@ pub mod insert {
 
                 (pos, pos, local_offs)
             };
-
-            let new_range = if doc.restore_cursor {
+            let new_range = if range.head > range.anchor {
```

Comment:
This will also extend if you enter insert mode via `o` (`open_below`), `O` (`open_above`), or mouse click when in insert mode and then hit enter. Instead I believe we want to check `range.cursor(text) > range.anchor` to avoid cases where the range is a point (`Range::point`)

Reply:
Done. However, I am not too familiar with `range.cursor(text)`